### PR TITLE
Make it more clear which module is being tested when running cargo test

### PR DESCRIPTION
**Pull request by guswynn**
I recently asked in zulip if this is a good idea, as I find it hard to find the module thats being tested from the complex path with the hash. 

Output of `cargo test`:
```
2021-02-21 13:29:33 I > ~/repos/cargo/target/debug/cargo test
    Finished test [unoptimized + debuginfo] target(s) in 0.42s
     Running unittests (target/debug/deps/test_tests-759130ea61f71571)

running 1 test
test tests::unit_test ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in
0.00s

     Running tests2/lib.rs (target/debug/deps/integration_tests-6b7f9fcd1721f083)

running 2 tests
test tests2_lib ... ok
test second_test::tests2_second ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in
0.00s

     Running tests/lib.rs (target/debug/deps/lib-d2be6d29597c2790)

running 2 tests
test second_test::tests_i_am_run_twice ... ok
test tests_lib ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in
0.00s
...
```

### CODE REVIEW

File: src/cargo/ops/cargo_test.rs [hunk 1]
```diff
@@ -85,7 +85,24 @@ fn run_unit_tests(
...
-        let exe_display = path.strip_prefix(cwd).unwrap_or(path).display();
+
+        let test_path = unit.target.src_path().path().unwrap();
+        let exe_display = if let TargetKind::Test = unit.target.kind() {
+            format!(
+                "{} ({})",
+                test_path
+                    .strip_prefix(unit.pkg.root())
+                    .unwrap_or(&test_path)
+                    .display(),
```
Comment (guswynn) [inline src/cargo/ops/cargo_test.rs:96]: this returns an opaque struct that implements Display, which I think makes it harder to collapse these 2 `format!`'s into 1 unless I `to_string()` this call and the `"unittests"` &has str, let me know what people would prefer

Comment (alexcrichton) [reply src/cargo/ops/cargo_test.rs:96]: Nah this seems fine I think, no need to worry too much about perf here!

**Comment by alexcrichton** [comment]
@rfcbot fcp merge

Thanks for the PR! This looks good to me and I think it's pretty reasonable!

I've cc'd other folks on the Cargo team to get their thoughts too since this is changing the output of Cargo by default.

**Comment by joshtriplett** [comment]
This looks great, thank you! I'm glad to see test output being made more helpful.

@rfcbot reviewed

**Pull request merged**